scanpy/plotting/_anndata.py

```diff
@@ -1349,14 +1355,28 @@ def dotplot(adata, var_names, groupby=None, use_raw=None, log=False, num_categor
     if use_raw is None and adata.raw is not None: use_raw = True
     if isinstance(var_names, str):
         var_names = [var_names]
-    categories, obs_tidy = _prepare_dataframe(adata, var_names, groupby, use_raw, log, num_categories, layer=layer)
+    categories, obs_tidy, categorical = _prepare_dataframe(adata, var_names, groupby, use_raw, log, num_categories, layer=layer)
 
     # for if category defined by groupby (if any) compute for each var_name
     # 1. the mean value over the category
     # 2. the fraction of cells in the category having a value > 0
 
     # 1. compute mean value
-    mean_obs = obs_tidy.groupby(level=0).mean()
+    if use_weights:
+        weights.columns = ['Wt']
```

Comment:
Mutating an input parameter is not a good idea. What if the user wants to use that object for something else later?

Reply:
Hmmm :) I made a copy and modified the copy locally.

Reply:
`df_weights = weights` doesn’t make a copy in Python. You need to either do `df_weights = some_df.copy()` or better something like `pd.DataFrame(dict(Wt=weights))`. This will allow users to pass in any sequence and will create a copy.